galasa-inttests-parent/dev.galasa.inttests/src/main/java/dev/galasa/inttests/compilation/AbstractCompilationLocal.java

```diff
@@ -0,0 +1,411 @@
+ * 
+ * (c) Copyright IBM Corp. 2021.
+ */
+package dev.galasa.inttests.compilation;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.Charset;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import org.apache.commons.logging.Log;
+import org.apache.http.ConnectionClosedException;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+
+import dev.galasa.BeforeClass;
+import dev.galasa.Test;
+import dev.galasa.core.manager.Logger;
+import dev.galasa.core.manager.RunName;
+import dev.galasa.core.manager.TestProperty;
+import dev.galasa.framework.spi.ResourceUnavailableException;
+import dev.galasa.galasaecosystem.IGenericEcosystem;
+import dev.galasa.http.HttpClient;
+import dev.galasa.http.IHttpClient;
+import dev.galasa.ipnetwork.IpNetworkManagerException;
+import dev.galasa.java.ubuntu.IJavaUbuntuInstallation;
+import dev.galasa.linux.ILinuxImage;
+import dev.galasa.linux.LinuxManagerException;
+
+public abstract class AbstractCompilationLocal {
+    
+    @RunName
+    public String         runName;
+    
+    @Logger
+    public Log              logger;
+    
+    @HttpClient
+    public IHttpClient      client;
+    
+    private String          prefix = "dev.galasa.simbank";
+    private String          testProjectName = prefix + ".tests";
+    private String          managerProjectName = prefix + ".manager";
+    
+    private Path            simplatformParentDir;
+    private Path            gradleBin;
+    
+    @TestProperty(prefix = "gradle.zip",suffix = "location", required = true)
+    public String           gradleZipLocation;
+    
+    public String           gradleZipVersion;
+    
+    String                  javaHomeCommand;
+    
+    /*
+     * Initialise all resources needed for testing.
+     * 
+     */
+    @BeforeClass
+    public void setupTest() throws ResourceUnavailableException, IOException, LinuxManagerException, IpNetworkManagerException {
+        javaHomeCommand = "export JAVA_HOME=" + getJavaInstallation().getJavaHome();
+        
+        gradleZipVersion = getGradleVersion();
+        simplatformParentDir = setupSimPlatform();
+        gradleBin = installGradle();
+    }
+    
+    /*
+     * Extracts the version from the gradle zip path
+     */
+    
+    public String getGradleVersion() {
+        final String regex = "gradle-(\\d+\\.\\d+\\.\\d+)-bin\\.zip";
+
+        final Pattern pattern = Pattern.compile(regex, Pattern.MULTILINE);
+        final Matcher matcher = pattern.matcher(gradleZipLocation);
```

Comment:
Should not be getting the version from the file name.    We will probably copy this to an internal repo.
You should be getting the gradle version by running the gradle command

Reply:
Just had a chat to Will,     think we need a Gradle Manager,   please create